yarn-project/noir-libs/noir-aztec/src/note/note_getter.nr

```diff
@@ -72,13 +72,13 @@ fn get_notes<Note, N, S, P>(
     let opt_notes = get_notes_internal(storage_slot, note_interface, options);
     for i in 0..opt_notes.len() {
         let opt_note = opt_notes[i];
-        let mut unique_note_hash = 0;
+        let mut note_hash_for_read_request = 0;
         if opt_note.is_some() {
             let note = opt_note.unwrap_unchecked();
             check_note_header(*context, storage_slot, note_interface, note);
-            unique_note_hash = compute_unique_note_hash(note_interface, note);
+            note_hash_for_read_request = compute_note_hash_for_read_request(note_interface, note);
         };
-        context.push_read_request(unique_note_hash);
+        context.push_read_request(note_hash_for_read_request);
```

Comment:
We should add tests to ensure that an oracle which maliciously returns a `note_header.nonce = 0` for a 'pre-existing note' results in a failing tx, because the resulting nullifier would be different from the 'correct' nullifier. Presumably the tx would correctly fail, either because the read request would result in a note which doesn't exist in the tree, or because the read request's note note would look like a 'pending note', but there wouldn't be a corresponding earlier function which created such a pending note.
Not sure where in the stack such a test should be orchestrated?

Reply:
https://github.com/AztecProtocol/aztec-packages/issues/1410

Reply:
Created ticket to add these tests ^

Reply:
and added TODO